torch/utils/data/dataloader.py

```diff
@@ -144,8 +145,9 @@ def __len__(self):
         return int(math.ceil(len(self.sampler) / float(self.batch_size)))
 
     def __next__(self):
-        if self.num_workers == 0:
-            # same-process loading
+        if self.num_workers == 0:  # same-process loading
+            if self.drop and self.samples_remaining < self.batch_size:
```

Comment:
It should be fixed in this PR. Otherwise the docs will be incorrect, because the flag will be completely ignored in the (btw most commonly used) multiprocess case.